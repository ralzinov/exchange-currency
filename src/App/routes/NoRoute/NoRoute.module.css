.host {
    display: flex;
    flex: 1;
    justify-content: center;
    flex-direction: column;
    align-items: center;
}
