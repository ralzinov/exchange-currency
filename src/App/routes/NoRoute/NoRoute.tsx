import * as React from 'react';
import {Link} from 'react-router-dom';
import styles from './NoRoute.module.css';

const NoRoute: React.FC = () => {
    return (
        <div className={styles.host}>
            <h1>404</h1>
            <Link to={'/'}>Go to main</Link>
        </div>
    );
};

export default NoRoute;
