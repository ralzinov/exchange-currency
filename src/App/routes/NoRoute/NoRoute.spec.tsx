import React from 'react';
import ReactDOM from 'react-dom';
import NoRoute from './NoRoute';
import {create} from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><NoRoute /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('should render with props and children', () => {
    const button = create(
        <BrowserRouter><NoRoute /></BrowserRouter>
    );
    expect(button.toJSON()).toMatchSnapshot();
});
