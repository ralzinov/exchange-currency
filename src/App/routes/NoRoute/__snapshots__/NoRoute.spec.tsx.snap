// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with props and children 1`] = `
<div
  className="host"
>
  <h1>
    404
  </h1>
  <a
    href="/"
    onClick={[Function]}
  >
    Go to main
  </a>
</div>
`;
